syntax = "proto3";

package orders;

option go_package = "trading-desk/internal/protos/orders";

// OrderRequest represents a request to place a trading order
message OrderRequest {
  string symbol = 1;          // Stock symbol (e.g., "AAPL")
  string qty = 2;             // Quantity as string to support decimals
  string side = 3;            // "buy" or "sell"
  string order_type = 4;      // "market", "limit", "stop", "stop_limit"
  string time_in_force = 5;   // "day", "gtc", "ioc", "fok"
  string limit_price = 6;     // Optional: limit price for limit orders
  string stop_price = 7;      // Optional: stop price for stop orders
}

// OrderResponse represents the response after placing an order
message OrderResponse {
  string status = 1;          // "success" or "error"
  string order_id = 2;        // Alpaca order ID
  string message = 3;         // Optional error message or additional info
  string symbol = 4;          // Echo back the symbol
  string qty = 5;             // Echo back the quantity
  string side = 6;            // Echo back the side
  string filled_qty = 7;      // Quantity filled so far
  string order_status = 8;    // Alpaca order status: "new", "filled", "partially_filled", etc.
}
